src/main/java/tech/stackable/gis/hbase/coprocessor/Neighbor.java

```diff
@@ -4,10 +4,12 @@ public class Neighbor {
     public final String key;
     public final double lat;
     public final double lon;
+    public final double distance;
 
-    public Neighbor(String key, double lat, double lon) {
```

Comment:
Now this class is structurally identical to the Point proto. Another (premature) optimization could be to get read of it and thus create less objects for the endpoint response.

Reply:
good Point :) See KNN.KNNResponse.newBuilder().addAllPoints(knn)